ambry-router/src/test/java/com.github.ambry.router/OperationTrackerTest.java

```diff
@@ -944,19 +994,64 @@ private void initialize() {
     mockClusterMap = new MockClusterMap(false, datanodes, 1, Collections.singletonList(mockPartition), localDcName);
   }
 
+  /**
+   * Initialize 4 DCs, 2 disk datacenters, 2 cloud datacenters. Each disk datacenter has 3 replicas, and each cloud
+   * datacenter has 1 replica.
+   * @param makeCloudDcLocal {@code true} to make the local datacenter one of the cloud datacenters.
+   */
+  private void initializeWithCloudDcs(boolean makeCloudDcLocal) {
+    List<Port> portList = Collections.singletonList(new Port(PORT, PortType.PLAINTEXT));
+    List<String> mountPaths = Collections.singletonList("mockMountPath");
+    mockPartition = new MockPartitionId();
+    List<MockDataNodeId> diskNodes = Arrays.asList(new MockDataNodeId(portList, mountPaths, "dc-0"),
+        new MockDataNodeId(portList, mountPaths, "dc-1"));
+    populateReplicaList(3 * diskNodes.size(), ReplicaState.STANDBY, diskNodes);
+    List<MockDataNodeId> cloudNodes = Arrays.asList(new MockDataNodeId(portList, Collections.emptyList(), "cloud-dc-0"),
+        new MockDataNodeId(portList, Collections.emptyList(), "cloud-dc-1"));
+    // only one cloud replica per cloud dc.
+    populateReplicaList(cloudNodes.size(), ReplicaState.STANDBY, cloudNodes);
+    datanodes = new ArrayList<>();
+    datanodes.addAll(diskNodes);
+    datanodes.addAll(cloudNodes);
+    localDcName = (makeCloudDcLocal ? cloudNodes : diskNodes).get(0).getDatacenterName();
+    mockClusterMap = new MockClusterMap(false, datanodes, 1, Collections.singletonList(mockPartition), localDcName);
+  }
+
   /**
    * Populate replicas for a partition.
    * @param replicaCount The number of replicas to populate.
    * @param replicaState The {@link ReplicaState} associated with these replicas.
    */
   private void populateReplicaList(int replicaCount, ReplicaState replicaState) {
+    populateReplicaList(replicaCount, replicaState, datanodes);
+  }
+
+  /**
+   * Populate replicas for a partition.
+   * @param replicaCount The number of replicas to populate.
+   * @param replicaState The {@link ReplicaState} associated with these replicas.
+   * @param datanodes the datanodes to populate with replicas
+   */
+  private void populateReplicaList(int replicaCount, ReplicaState replicaState, List<MockDataNodeId> datanodes) {
     for (int i = 0; i < replicaCount; i++) {
       ReplicaId replicaId = new MockReplicaId(PORT, mockPartition, datanodes.get(i % datanodes.size()), 0);
       mockPartition.replicaIds.add(replicaId);
       mockPartition.replicaAndState.put(replicaId, replicaState);
     }
   }
 
+  /**
+   * @param replicaType the type of replica to filter by.
+   * @return the datacenter names with replicas of type {@code replicaType}.
+   */
+  private Set<String> getDatacenters(ReplicaType replicaType) {
+    return mockPartition.getReplicaIds()
+        .stream()
+        .filter(r -> r.getReplicaType() == ReplicaType.CLOUD_BACKED)
```

Comment:
Should compare to replicaType, no?

Reply:
Yes, will fix this in my commit to add more tests and address comments.